owlkettle/autoform.nim

```diff
@@ -0,0 +1,135 @@
+import std/[options, json, times, macros, strutils, sequtils]
+import ./dataentries
+import ./adw
+import ./widgetutils
+import ./guidsl
+import ./widgetdef
+import ./widgets
+
+macro getField*(someType: untyped, fieldName: static string): untyped =
+  nnkDotExpr.newTree(someType, ident(fieldName))
+
+proc `%`*(t: tuple): JsonNode =
```

Comment:
Remove, see below.

Reply:
Will do (or not do) based on the outcome of our discussion around 
`proc toFormField(state: auto, fieldName: static string, typ: typedesc[object | ref object | tuple | seq]): Widget =`